### PR TITLE
Hunting allocations 2

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes

- Add statics for [0] and [1].
- Storage slots returned as spans.
- Decode accounts from spans
- More ReadOnlySpan and ReadOnlyMemory

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by benaadams — APPROVED** [review]

File: src/Nethermind/Nethermind.Trie/PatriciaTree.cs [hunk 1]
```diff
@@ -355,7 +355,7 @@ private void SetRootHash(Hash256? value, bool resetObjects)
 
         [SkipLocalsInit]
         [DebuggerStepThrough]
-        public virtual byte[]? Get(ReadOnlySpan<byte> rawKey, Hash256? rootHash = null)
+        public virtual Span<byte> Get(ReadOnlySpan<byte> rawKey, Hash256? rootHash = null)
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Trie/PatriciaTree.cs:358]: Why not ReadOnlySpan?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Trie/PatriciaTree.cs]: yeah, why not

File: src/Nethermind/Nethermind.Evm/VirtualMachine.cs [hunk 2]
```diff
@@ -2747,10 +2740,10 @@ private static EvmExceptionType InstructionLog<TTracing>(EvmState vmState, ref E
 
         if (typeof(TTracingInstructions) == typeof(IsTracing))
         {
-            Span<byte> valueToStore = newIsZero ? BytesZero : bytes;
-            bytes = new byte[32]; // do not stackalloc here
-            storageCell.Index.ToBigEndian(bytes);
-            _txTracer.ReportStorageChange(bytes, valueToStore);
+            ReadOnlySpan<byte> valueToStore = newIsZero ? BytesZero.AsSpan() : bytes;
+            byte[] storageBytes = new byte[32]; // do not stackalloc here
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm/VirtualMachine.cs:2744]: I'm urged to stackalloc

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Evm/VirtualMachine.cs:2744]: https://github.com/NethermindEth/nethermind/pull/937/files#diff-9742ac85df13f5104064b9cf775ea804cc783bc16b9655d45c55774b13be19e6R2001

**Review by asdacap — APPROVED** [review]

**Pull request merged**